Add Connections, Secrets and Variables to Airflow Helm Chart documentation (#16059)
mistercrunch/Airflow: docs/helm-chart/adding-connections-and-variables.rst

 .. Licensed to the Apache Software Foundation (ASF) under one
    or more contributor license agreements.  See the NOTICE file
    distributed with this work for additional information
    regarding copyright ownership.  The ASF licenses this file
    to you under the Apache License, Version 2.0 (the
    "License"); you may not use this file except in compliance
    with the License.  You may obtain a copy of the License at

 ..   http://www.apache.org/licenses/LICENSE-2.0

 .. Unless required by applicable law or agreed to in writing,
    software distributed under the License is distributed on an
    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, either express or implied.  See the License for the
    specific language governing permissions and limitations
    under the License.

> Adding Connections, Secrets and Environment Variables

Reviewer: ```suggestion
Adding Connections, Variables and Environment Variables
```

I wouldn't consider `Secrets` to be core concept 🤷‍♂️

> Adding Connections, Secrets and Environment Variables
> ==================================================

Reviewer 2: ```suggestion
Adding Connections, Secrets and Environment Variables
=====================================================
```

You can programmatically add Connections, Secrets and Environment Variables to your

> airflow deployment using Helm chart.

Reviewer: ```suggestion
You can programmatically add Connections, Variables and arbitrary Environment Variables to your
Airflow deployment using the Helm chart.
```

I'd probably leave the k8s secret piece out of the mix here, personally.

> Connections and Secrets

Reviewer: ```suggestion
Connections and Sensitive Environment Variables
```

Maybe?

-----------------------
Under the ``secret`` and ``extraSecret`` sections of the ``values.yaml`` you can pass connection strings and secrets

> into Airflow using Helm chart. To illustrate, lets create a yaml file to override values under these sections of the

Reviewer: ```suggestion
Under the ``secret`` and ``extraSecret`` sections of the ``values.yaml`` you can pass connection strings and sensitive environemt variables
into Airflow using the Helm chart. To illustrate, lets create a yaml file to override values under these sections of the
```

``values.yaml`` file.

.. code-block:: yaml

   secret:
      - envName: "my-env"
        secretName: "my-secret-name"
        secretKey: "my-secret-key"

  extraSecrets:
     {{ .Release.Name }}-airflow-connections:
       data: |
         AIRFLOW_CONN_GCP: 'base64_encoded_gcp_conn_string'
         AIRFLOW_CONN_AWS: 'base64_encoded_aws_conn_string'
       stringData: |
         AIRFLOW_CONN_OTHER: 'other_conn'
     {{ .Release.Name }}-other-secret-name-suffix: |

>        data: |

Reviewer: ```suggestion
   secret:
     - envName: "AIRFLOW_CONN_GCP"
        secretName: "my-airflow-connections"
        secretKey: "AIRFLOW_CONN_GCP"
      - envName: "my-env"
        secretName: "my-secret-name"
        secretKey: "my-secret-key"

  extraSecrets:
     my-airflow-connections:
       data: |
         AIRFLOW_CONN_GCP: 'base64_encoded_gcp_conn_string'
     my-secret-name: |
       stringData: |
         my-secret-key: my-secret
```

Have it be a working example, and show both an env var and a connection.

Variables
---------

> Airflow supports Variables which enable users craft dynamic DAGs. You can set variables in Airflow in three ways - UI,

Reviewer: ```suggestion
Airflow supports Variables which enable users to craft dynamic DAGs. You can set Variables in Airflow in three ways - UI,
```

> commandline, and within your DAG file. See:doc:`/howto/variable`.` for more.

Reviewer: ```suggestion
commandline, and within your DAG file. See :doc:`/howto/variable` for more.
```
I think this is the right format for the link?

> With Helm chart, you can also inject environment variables into airflow. For example, in a yaml file to

Reviewer: ```suggestion
With the Helm chart, you can also inject environment variables into Airflow. For example, in a yaml file to
```

override values of interest in the ``env`` section of the ``values.yaml`` file.

.. code-block:: yaml

   env:
     - name: "AIRFLOW_VAR_KEY"
       value: "value_1"

> 

Reviewer: ```suggestion
```

     - name: "AIRFLOW_VAR_ANOTHER_KEY"
       value: "value_2"

You can also utilize the ``extraEnv`` and ``extraEnvFrom`` to set more possibly "templatable" values for your airflow

> deployment.

Reviewer: ```suggestion
You can also utilize ``extraEnv`` and ``extraEnvFrom`` if you need the name or value to be templated.
```

Reviewer: Bump, was this accidentally missed?

.. code-block:: yaml

   extraEnv: |
     - name: AIRFLOW__CORE__LOAD_EXAMPLES

>        value: 'True'

Reviewer: ```suggestion
     - name: AIRFLOW_VAR_HELM_RELEASE_NAME
       value: '{{ .Release.Name }}'
```

Might as well have an example that uses templating.

   extraEnvFrom: |
     - secretRef:
         name: '{{ .Release.Name }}-airflow-connections'
     - configMapRef:

>          name: '{{ .Release.Name }}-airflow-variables'

Reviewer: ```suggestion
   extraEnvFrom: |
     - configMapRef:
         name: '{{ .Release.Name }}-airflow-variables'
         
   extraConfigMaps:
     '{{ .Release.Name }}-airflow-variables':
       data: |
         AIRFLOW_VAR_HELLO_MESSAGE: "Hi!"
```
Might as well be a complete working example.

You can programmatically add Connections, Variables and arbitrary Environment Variables to your
Airflow deployment using the Helm chart.

Connections and Sensitive Environment Variables
-----------------------------------------------
Under the ``secret`` and ``extraSecret`` sections of the ``values.yaml`` you can pass connection strings and sensitive
environment variables into Airflow using the Helm chart. To illustrate, lets create a yaml file called ``override.yaml``
to override values under these sections of the ``values.yaml`` file.

.. code-block:: yaml

   # override.yaml

   secret:
     - envName: "AIRFLOW_CONN_GCP"
        secretName: "my-airflow-connections"
        secretKey: "AIRFLOW_CONN_GCP"
      - envName: "my-env"
        secretName: "my-secret-name"
        secretKey: "my-secret-key"

   extraSecrets:
     my-airflow-connections:
       data: |
         AIRFLOW_CONN_GCP: 'base64_encoded_gcp_conn_string'
     my-secret-name: |
       stringData: |
         my-secret-key: my-secret

Variables
---------
Airflow supports Variables which enable users to craft dynamic DAGs. You can set Variables in Airflow in three ways - UI,

> commandline, and within your DAG file. See :doc:`/howto/variable` for more.

Reviewer: ```suggestion
cli, and within your DAG file. See :doc:`apache-airflow:howto/variable` for more.
```

Should fix the spelling and link issue.

        secretKey: "AIRFLOW_CONN_GCP"
      - envName: "my-env"
        secretName: "my-secret-name"
        secretKey: "my-secret-key"

   extraSecrets:
     my-airflow-connections:
       data: |
         AIRFLOW_CONN_GCP: 'base64_encoded_gcp_conn_string'
     my-secret-name: |
       stringData: |
         my-secret-key: my-secret

Variables
---------
Airflow supports Variables which enable users to craft dynamic DAGs. You can set Variables in Airflow in three ways - UI,
commandline, and within your DAG file. See :doc:`/howto/variable` for more.

With the Helm chart, you can also inject environment variables into Airflow. So in the example ``override.yaml`` file,
we can override values of interest in the ``env`` section of the ``values.yaml`` file.

.. code-block:: yaml

   env:
     - name: "AIRFLOW_VAR_KEY"
       value: "value_1"
     - name: "AIRFLOW_VAR_ANOTHER_KEY"
       value: "value_2"

You can also utilize the ``extraEnv`` and ``extraEnvFrom`` to set more possibly "templatable" values for your airflow
deployment.

.. code-block:: yaml

   extraEnv: |
     - name: AIRFLOW_VAR_HELM_RELEASE_NAME
       value: '{{ .Release.Name }}'

> 

Reviewer: ```suggestion
```
Less vertical space.
